Source/LinuxParking.API/Controllers/SpotController.cs

```diff
@@ -1,49 +1,108 @@
 using System;
+using AutoMapper;
 using System.Threading.Tasks;
-using System.Collections.Generic;
 using Microsoft.AspNetCore.Mvc;
+using LinuxParking.API.Extentions;
+using LinuxParking.API.Domain.Resources;
+using LinuxParking.API.Domain.Interfaces.Services;
+using LinuxParking.API.Domain.Models;
+using System.Collections.Generic;
 
 namespace LinuxParking.API.Controllers
 {
     [Route("/api/station/{stationId}/[controller]")]
     public class SpotController : Controller
     {
-        public SpotController()
+        private readonly ISpotService _spotService;
+        private readonly IMapper _mapper;
+
+        public SpotController(ISpotService spotService, IMapper mapper)
         {
-            // TODO: Dependecy injections
+            _spotService = spotService;
+            _mapper = mapper;
         }
 
         [HttpPost]
-        public async Task CreateAsync([FromRoute] string stationId)
+        public async Task<IActionResult> CreateAsync([FromRoute] int stationId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            spot.StationId = stationId;
+            var res = await _spotService.SaveAsync(spot);
+
+            if (!res.Success)
+            {
+                return BadRequest(res.Message);
+            }
+
+            var spotResource = _mapper.Map<Spot, SpotResource>(res.Spot);
+            return Created("", spotResource);
         }
 
         [HttpGet]
-        public async Task GetAllAsync([FromRoute] string stationId)
+        public async Task<IActionResult> GetAllAsync([FromRoute] int stationId)
         {
-            throw new NotImplementedException();
+            var res = await _spotService.ListAsync(stationId);
+
+            if (!res.Success)
```

Comment:
We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```